java-frontend/src/main/java/org/sonar/java/ast/visitors/FileLinesVisitor.java

```diff
@@ -89,4 +219,30 @@ public void visitToken(SyntaxToken syntaxToken) {
       }
     }
   }
+
+  private static boolean isConstant(VariableTree variableTree) {
+    return ModifiersUtils.hasModifier(variableTree.modifiers(), Modifier.STATIC)
```

Comment:
What about fields from interfaces, which are implicitly static and final, and does not necessarily have the modifier?